### PR TITLE
Render preview based of entity domain

### PR DESCRIPTION
**Pull request by gjohansson-ST**
## Proposed change

Render preview based off the domain of the entity (currently all looks like a sensor).

I think probably needs to walk this though again to make sure all relevant ones are added (and possibly maybe remove something I could guess). Not all can be tested as there is limited preview entities currently.

Examples:
image
image
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
	- Improved rendering logic for various entity states in the Home Assistant interface.
	- Added support for additional entity types, enhancing user interaction options.
- **Style**
	- Introduced new CSS styles for better layout and appearance of rendered components, including flexbox styles for improved alignment.
- **Refactor**
	- Centralized rendering logic for different entity domains, improving code organization and maintainability.

### CODE REVIEW

File: src/dialogs/config-flow/previews/entity-preview-row.ts [hunk 1]
```diff
@@ -58,8 +62,252 @@ class EntityPreviewRow extends LitElement {
...
+        isUnavailableState(stateObj.state);
+      return html`
+        ${showToggle
+          ? html`
+              <ha-entity-toggle
+                .hass=${this.hass}
+                .stateObj=${stateObj}
+              ></ha-entity-toggle>
+            `
+          : html` ${this.hass.formatEntityState(stateObj)} `}
```
Comment (silamon) [inline src/dialogs/config-flow/previews/entity-preview-row.ts:133]:
```suggestion
          : this.hass.formatEntityState(stateObj)}
```

File: src/dialogs/config-flow/previews/entity-preview-row.ts [hunk 2]
```diff
@@ -58,8 +62,252 @@ class EntityPreviewRow extends LitElement {
...
+      return html`
+        ${showSensor
+          ? html`
+              <hui-timestamp-display
+                .hass=${this.hass}
+                .ts=${new Date(stateObj.state)}
+                capitalize
+              ></hui-timestamp-display>
+            `
+          : html` ${this.hass.formatEntityState(stateObj)} `}
```
Comment (silamon) [inline src/dialogs/config-flow/previews/entity-preview-row.ts:117]:
```suggestion
          : this.hass.formatEntityState(stateObj)}
```

File: src/dialogs/config-flow/previews/entity-preview-row.ts [hunk 3]
```diff
@@ -58,8 +62,252 @@ class EntityPreviewRow extends LitElement {
...
+        <div>
+          ${isUnavailableState(stateObj.state) ||
+          stateObj.attributes.temperature === undefined ||
+          stateObj.attributes.temperature === null
+            ? this.hass.formatEntityState(stateObj)
+            : this.hass.formatEntityAttributeValue(stateObj, "temperature")}
+        </div>
+      `;
+    }
+    return html` ${this.hass.formatEntityState(stateObj)} `;
```
Comment (silamon) [inline src/dialogs/config-flow/previews/entity-preview-row.ts:309]:
```suggestion
    return this.hass.formatEntityState(stateObj);
```

**Review by silamon — COMMENTED** [review]
No need to use html whenever you are just returning a string.

**Review by silamon — DISMISSED** [review]

**Comment by silamon** [comment]
> I think probably needs to walk this though again to make sure all relevant ones are added (and possibly maybe remove something I could guess). Not all can be tested as there is limited preview entities currently.

Not sure if you have already done this at this time, I've went through the code again and tested. 
I think it can be merged and improved later on.

**Comment by gjohansson-ST** [comment]
> Not sure if you have already done this at this time, I've went through the code again and tested. I think it can be merged and improved later on.

I got stuck on trying to implement `image` and then lost this a bit.
Let me rebase and sort them and we can bring this in 👍

**Comment by gjohansson-ST** [comment]
@silamon I think this should be ready to go now 👍

**Review by MindFreeze — APPROVED** [review]

**Pull request merged**